### PR TITLE
Init scripts to install cuda11 runtime [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Add the parameter 'init_script' to make init scripts common for Databricks.

Add the 'init_cuda11_runtime.sh' to install cuda11 toolkit on Databricks.

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

File: jenkins/databricks/clusterutils.py [hunk 1]
```diff
@@ -46,13 +46,18 @@ def generate_create_templ(sshKey, cluster_name, runtime, idle_timeout,
...
         templ['ssh_public_keys'] = [ sshKey ]
         templ['num_workers'] = num_workers
-        templ['init_scripts'] = [
-            {
-                "dbfs": {
-                     "destination": "dbfs:/databricks/init_scripts/init_cudf_udf.sh"
-                }
-            }
-        ]
+        if (init_script != ''):
```
Comment (NvTimLiu) [inline jenkins/databricks/clusterutils.py:49]: We'll add -f dbfs:/databricks/init_scripts/init_cudf_udf.sh,dbfs:/databricks/init_scripts/init_cuda11_runtime.sh in Jenkinsfile.databricks301nightly to init cudf_udf and cuda11

File: jenkins/databricks/create.py [hunk 2]
```diff
@@ -34,14 +34,17 @@ def main():
   worker_type = 'g4dn.xlarge'
   driver_type = 'g4dn.xlarge'
   cloud_provider = 'aws'
+  # comma seperated init scritps, e.g., dbfs:/foo,dbfs:/bar,...
+  init_script = ''
```
Comment (NvTimLiu) [inline jenkins/databricks/create.py:38]: It's easy to extend the init scripts in the Jenkinsfile.

Comment (tgravescs) [inline jenkins/databricks/create.py:38]: nit: rename to be init_scripts (ie with an s) everywhere

File: jenkins/databricks/init_cuda11_runtime.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# The init script to install cuda11.0 toolkit
+# Will be automatically pushed into the dbfs:/databricks/init_scripts once it is updated.
+
+wget http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux.run
+
+sudo sh cuda_11.0.2_450.51.05_linux.run --toolkit --silent
```
Comment (tgravescs) [inline jenkins/databricks/init_cuda11_runtime.sh:23]: so does this make the runtime show up in apt as well?   I assume based on the --toolkit parameter it only installed toolkit and cuda 11.0 runtime and the driver version status the same? (I haven't used this method before so just double checking).  Any difference between this and the instructions I sent as far as runtime?

**Comment by NvTimLiu** [comment]
build

File: jenkins/databricks/init_cuda11_runtime.sh (hunk 3 above)
Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh:23]: > so does this make the runtime show up in apt as well? 
Sorry, what does `show up in apt` mean?

Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh:23]: > I assume based on the --toolkit parameter it only installed toolkit and cuda 11.0 runtime and the driver version status the same? 
Correct

Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh:23]: > Any difference between this and the instructions I sent as far as runtime?

Should be the same,  Both `*.run[local]`  and `deb[local]`  download `toolkit + dirvers` tar to local, untar and install.

`*.run[local]`   CLI seems simple and clean.

**Comment by NvTimLiu** [comment]
We can close the dup PR: https://github.com/NVIDIA/spark-rapids/pull/2179 later

File: jenkins/databricks/init_cuda11_runtime.sh (hunk 3 above)
Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh:23]: > > so does this make the runtime show up in apt as well?
> > Sorry, what does `show up in apt` mean? Do you mean when installing, are there any dialog showing up to let user select what to install(cuda toolkit/dirver/examples)? The pop up dialog is disabled by `--silence`

File: jenkins/databricks/create.py (hunk 2 above)
Comment (NvTimLiu) [reply jenkins/databricks/create.py]: fixed

**Comment by NvTimLiu** [comment]
build

File: jenkins/databricks/create.py [hunk 4]
```diff
@@ -34,14 +34,17 @@ def main():
...
+  # comma seperated init scritps, e.g., dbfs:/foo,dbfs:/bar,...
+  init_scripts = ''
+
 
   try:
-      opts, args = getopt.getopt(sys.argv[1:], 'hw:t:k:n:i:r:o:d:e:s:',
+      opts, args = getopt.getopt(sys.argv[1:], 'hw:t:k:n:i:r:o:d:e:s:f:',
                                  ['workspace=', 'token=', 'sshkey=', 'clustername=', 'idletime=',
-                                     'runtime=', 'workertype=', 'drivertype=', 'numworkers=', 'cloudprovider='])
+                                     'runtime=', 'workertype=', 'drivertype=', 'numworkers=', 'cloudprovider=, initscripts='])
```
Comment (GaryShen2008) [inline jenkins/databricks/create.py:44]: 'cloudprovider=, initscripts='
Should it be 'cloudprovider=', 'initscripts='?

Comment (NvTimLiu) [reply jenkins/databricks/create.py]: fixed

**Comment by NvTimLiu** [comment]
build

File: jenkins/databricks/create.py [hunk 5]
```diff
@@ -34,14 +34,17 @@ def main():
   worker_type = 'g4dn.xlarge'
   driver_type = 'g4dn.xlarge'
   cloud_provider = 'aws'
+  # comma seperated init scritps, e.g., dbfs:/foo,dbfs:/bar,...
```
Comment (pxLi) [inline jenkins/databricks/create.py:37]: typo: 
scritps -> scripts
seperated -> separated
e.g., -> e.g.

File: jenkins/databricks/init_cuda11_runtime.sh (hunk 3 above)
Comment (pxLi) [inline jenkins/databricks/init_cuda11_runtime.sh:23]: why `sudo` here, is databricks default login user not root?

Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh:23]: correct, default is `ubuntu`

File: jenkins/databricks/create.py (hunk 5 above)
Comment (NvTimLiu) [reply jenkins/databricks/create.py]: fixed

**Review by pxLi — APPROVED** [review]
LGTM as internal tests passed

**Comment by NvTimLiu** [comment]
build

**Review by GaryShen2008 — DISMISSED** [review]

File: jenkins/databricks/init_cuda11_runtime.sh (hunk 3 above)
Comment (NvTimLiu) [reply jenkins/databricks/init_cuda11_runtime.sh]: > correct, default is `ubuntu`

Sorry, the init scripts user is `root`, remove sudo here

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
As pre-merge pipeline won't run against (include) this change, skip CI.

Test DB pipeline on Blossom Jenkins, and got PASS.

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**